pyccel/ast/f2py.py

```diff
@@ -127,7 +127,7 @@ def as_static_function(func, name=None):
             transpose_stmts = [Comment(f2py_template.format(kind    = int_kind,
                                                             index   = i,
                                                             name    = a.name,
-                                                            index_t = str(a.rank - i - 1)))
+                                                            index_t = str(i)))
```

Comment:
Why are we not reversing the shape anymore?

Reply:
See:
https://github.com/pyccel/pyccel/pull/286/files#r399179413

Reply:
At this point it looks like we do not need to transpose the indices for f2py (`transpose_stmts` variable), and maybe not even to declare the array as having C ordering (`c_intent_stmt` variable). Can you please try and remove each of them?

Reply:
If I remove both then the first test to fail is: `test_array_int32_2d_C_scalar_add`
If I remove just `transpose_stmts` then this passes but a different test fails. In this case the first to fail is : `test_array_int_2d_C_initialization`
If I remove just `c_intent_stmt` then the failing test is once again: `test_array_int32_2d_C_scalar_add`

Reply:
The error for `test_array_int32_2d_C_scalar_add `is : `ValueError: failed to initialize intent(inout) array -- input not fortran contiguous`
For `test_array_int_2d_C_initialization` the initialised values are wrong:
```
E       assert False
E        +  where False = <function array_equal at 0x7f78370b72f0>(array([[1, 2, 3],\n       [4, 5, 6]]), array([[  1,   2,   4],\n       [  5,  64, 721]]))
E        +    where <function array_equal at 0x7f78370b72f0> = np.array_equal
```

Reply:
So we really need those statements, and the tests are strict enough to identify any problem. Thanks for trying!